backport_packages/refactor_backport_packages.py

```diff
@@ -0,0 +1,703 @@
+            @@ -27,7 +27,7 @@
+             import jinja2
+             from cattr import structure, unstructure
+
+            -from airflow.models.base import Operator
+            +from airflow.providers.papermill.utils.base import Operator
+             from airflow.utils.module_loading import import_string
+
+             ENV = jinja2.Environment()
+
+        """
+        # noinspection PyUnusedLocal
+        def papermill_package_filter(node: LN, capture: Capture, filename: Filename) -> bool:
+            return filename.startswith("./airflow/providers/papermill/")
+
+        os.makedirs(os.path.join(get_target_providers_package_folder("papermill"), "utils", "lineage"),
+                    exist_ok=True)
+        copyfile(
+            os.path.join(get_source_airflow_folder(), "airflow", "utils", "__init__.py"),
+            os.path.join(get_target_providers_package_folder("papermill"), "utils", "__init__.py")
+        )
+        copyfile(
+            os.path.join(get_source_airflow_folder(), "airflow", "lineage", "__init__.py"),
+            os.path.join(get_target_providers_package_folder("papermill"), "utils", "lineage", "__init__.py")
+        )
+        copyfile(
+            os.path.join(get_source_airflow_folder(), "airflow", "lineage", "entities.py"),
+            os.path.join(get_target_providers_package_folder("papermill"), "utils", "lineage", "entities.py")
+        )
+        copyfile(
+            os.path.join(get_source_airflow_folder(), "backport_packages", "template_base_operator.py.txt"),
+            os.path.join(get_target_providers_package_folder("papermill"), "utils", "base.py")
+        )
+        (
+            self.qry.
+            select_module("airflow.lineage.entities").
+            filter(callback=papermill_package_filter).
+            rename("airflow.providers.papermill.utils.lineage.entities")
+        )
+        (
+            self.qry.
+            select_module("airflow.lineage").
+            filter(callback=papermill_package_filter).
+            rename("airflow.providers.papermill.utils.lineage")
+        )
+        # Papermill uses lineage which uses Operator under the hood so we need to change it as well
+        (
+            self.qry.
+            select_module("airflow.models.base").
+            filter(callback=papermill_package_filter).
+            rename("airflow.providers.papermill.utils.base")
+        )
+
+    def do_refactor(self, in_process: bool = False) -> None:
+        self.rename_deprecated_modules()
+        self.refactor_amazon_package()
+        self.refactor_google_package()
+        self.refactor_odbc_package()
+        self.refactor_papermill_package()
+        self.remove_tags()
+        self.remove_super_init_call()
+        self.add_provide_context_to_python_operators()
+        self.remove_poke_mode_only_decorator()
+        # In order to debug Bowler - set in_process to True
+        self.qry.execute(write=True, silent=False, interactive=False, in_process=in_process)
```

Comment:
Looks nice and this is definitely more readable 👌

Reply:
Cool. I think I fixed last failure. With this small thing on how to debug Bowler and having it as a switch on python execution, I think it is super important. I committed it a few times incidentally before and I wondered why it runs so slow :).